src/skin/legacyskinparser.cpp

```diff
@@ -1084,6 +1089,29 @@ QWidget* LegacySkinParser::parseCoverArt(QDomElement node) {
     return pCoverArt;
 }
 
+void LegacySkinParser::parseSingletonDefinition(QDomElement node) {
+    // The actual singleton definition is identical to any other WWidgetGroup.
+    QWidget* child_group = parseWidgetGroup(node);
+    WSingletonContainer::defineSingleton(child_group->objectName(),
```

Comment:
As written singletons are parented to the parent widget where they were defined in the skin hierarchy and when a singleton container claims it the widget is reparented -- so the widget can never leak since it always has a parent. It would be nice to save the work on creating the widget if a singleton is already defined but this is probably a minor case and an error on the part of the skin writer anyway.